[Migrator] Make sure to lowercase properties in SetterToProperty

bitjammer opened the pull request:
Previously we were only stripping `set` from the name and not
lowercasing the property.

rdar://problem/32845968

bitjammer committed:
[Migrator] Make sure to lowercase properties in SetterToProperty

Previously we were only stripping `set` from the name and not
lowercasing the property.

rdar://problem/32845968

bitjammer commented:
@swift-ci Please test

harlanhaskins commented on lib/Migrator/APIDiffMigratorPass.cpp:
```diff
@@ -605,8 +605,11 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
-                                                   substr(3)) + " = ").str());
+          auto Replacement = (llvm::Twine(Walker.Result.str()
+                                          .substr(3)) + " = ").str();
+          Replacement[0] = tolower(Replacement[0]);
```
It might be worth using `camel_case:: toLowercaseWord()`  from `swift/Basic/StringExtras.h` here, so we don't migrate `.getURL` to `.uRL`.

bitjammer replied:
Cool, does that work with `Current` -> `current`?

harlanhaskins replied:
Yep! It walks the string and lowercases all consecutive uppercase letters up to the letter before the first lowercase letter.

`URL` -> `url`
`Current` -> `current`
`URLString` -> `urlString`
`XMLHTTPRequest` -> `xmlhttpRequest` (can't win 'em all)

harlanhaskins replied:
But IIRC this is what the Clang Importer uses to import names with ObjC styling:
https://github.com/apple/swift/blob/master/lib/ClangImporter/ImportName.cpp#L1454

bitjammer replied:
Yeah I vaguely remember seeing it but I couldn't remember the behavior. I'll switch to that, thanks for the tip!

bitjammer committed:
[Migrator] Use toLowercaseWord to lowercase SetterToProperty changes

Post-commit review follow-up:
This works better for properties that were all-caps, such as `URL`.

Thanks @harlanhaskins for the tip!

rdar://problem/32845968

bitjammer commented:
@swift-ci Please test

bitjammer commented:
@swift-ci Please test
